### ISSUE TITLE
pixel normalization and NN weight inits

### ISSUE THREAD

**Issue by danbider** [mlmed/torchxrayvision#9]
Hi, 

Thanks for sharing these helpful dataset objects. I tried training another (shallower) architecture on these data with default weight inits and I start getting `inf`s and `nan`s after the first training iteration. When I normalized the pixels to have mean=0, std=1 then it worked fine. 
Did you encounter this problem? How did you initialize your weights to prevent things from blowing up?
Dan

**Comment by ieee8023** [comment]
Sorry for the delay I am so overwhelmed these days!

So [-1024,1024] is arbitrary but we didn't want to have issues with float round off and some images could be 16-bit. The goal of this was to capture the potential range of the image. 

A reason to not normalize based on the stdev is: lets say you have a 8 bit image and the values were just noise around 0 then this means something and you should keep the pixels values at that level. But if you scale it by the stdev then the pixel values would be huge even though the image had nothing in it. So having a fixed range that we project the image too make sense here. So for a 8 bit image you have the range 0,256 mapped to -1024,1024 and for 16-bit you have 0,65536 mapped to -1024,1024.

I didn't have any issues with the default torch init. Here is the training script I used: https://github.com/ieee8023/xray-generalization/blob/master/train-joe.py which is slowly being converted over to torchxrayvision.

**Comment by danbider** [comment]
OK -- thanks for the explanation! the training script is useful. 
The thing that solved my problem is to add batchnorm in each layer. It wasn't needed when the images were normalized, but now with this scale it seems crucial.

**Comment by ieee8023** [comment]
Closing this because I think the issue is resolved. Open it again if it is not!
